File: labs/lab4/app/src/main/java/com/example/lab4/MainActivity.java
package com.example.lab4;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.io.IOException;
import java.io.InputStream;

import name.ank.lab4.BibDatabase;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private BibLibAdapter bibLibAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        InputStream publications = getResources().openRawResource(R.raw.publications_ferro_en);

        recyclerView = findViewById(R.id.recycler_view);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);

        recyclerView.setHasFixedSize(true);

        try {
            bibLibAdapter = new BibLibAdapter(publications);
        } catch (IOException e) {
            e.printStackTrace();
        }
        recyclerView.setAdapter(bibLibAdapter);
    }
}

File: labs/lab7/PictureBroadcastReceiver/app/src/main/java/com/example/picturebroadcastreceiver/MainActivity.kt
package com.example.picturebroadcastreceiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    val intentAction = "com.example.pictureurldownloading.PICTURE_DOWNLOAD"
    lateinit var button: Button
    lateinit var textView: TextView
    private lateinit var broadcastReceiver: PictureBroadcastReceiver

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textView = findViewById(R.id.textView)
        button = findViewById(R.id.button)
        if (intent.getStringExtra("path") != null) {
            Log.i("Activity", "Text was switched")
            textView.text = intent?.getStringExtra("path").toString()
        }

        broadcastReceiver = PictureBroadcastReceiver()
        registerReceiver(broadcastReceiver, IntentFilter(intentAction))

        button.setOnClickListener {
            textView.text = "Путь сброшен"
        }
    }

    class PictureBroadcastReceiver : BroadcastReceiver() {
        override fun onReceive(context: Context?, intent: Intent?) {
            Log.i("Broadcast Receiver", "Message was receive")
            val intentMainActivity = Intent(context, MainActivity::class.java)
            intentMainActivity.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK
                    or Intent.FLAG_ACTIVITY_NEW_TASK)
            val path = intent?.getStringExtra("url").toString()
            intentMainActivity.putExtra("path", path)
            context?.startActivity(intentMainActivity)
        }
    }

    override fun onDestroy() {
        unregisterReceiver(broadcastReceiver)
        super.onDestroy()
    }
}